src/types.ts

```diff
@@ -45,16 +46,13 @@ export type Size = "sm" | "md" | "lg";
 
 export type SizeProp = Size | ResponsiveObject<Size> | Size[];
 
-export type TagVariant = "subtle" | "solid" | "outline" | (string & {});
+export type TagVariant = ThemeTypings["components"]["Tag"]["variants"];
 
 export type SelectedOptionStyle = "color" | "check";
 
-export type Variant =
-  | "outline"
-  | "filled"
-  | "flushed"
-  | "unstyled"
-  | (string & {});
+export type Variant = ThemeTypings["components"]["Input"]["variants"];
+
+export type ColorScheme = ThemeTypings["colorSchemes"];
```

Comment:
I realized I could get intellisense autocompletion for this by using Chakra's built in type for it. I also changed the other `Variant` types to use Chakra's built in types for these components instead of defining them manually.